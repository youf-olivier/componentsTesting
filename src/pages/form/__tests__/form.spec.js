import React from 'react';
import { render, screen } from '@testing-library/react';
import Component from '../form';

describe('Form test suite', () => {
  const users = [
    {
      id: 1,
      login: 'login1',
      avatar_url: 'http://avatar1.jpg',
      html_url: 'http://comptegithub.com',
    },
    {
      id: 2,
      login: 'login2',
      avatar_url: 'http://avatar2.jpg',
      html_url: 'http://comptegithub.com',
    },
    {
      id: 3,
      login: 'login3',
      avatar_url: 'http://avatar3.jpg',
      html_url: 'http://comptegithub.com',
    },
  ];

  const inputs = {
    githubAccount: {
      label: 'Compte github',
      value: 'test',
      id: 'githubAccount',
      message: 'erreur de saisie',
    },
    userName: {
      label: "Nom d'utilisateur",
      value: 'Jean',
      id: 'username',
      message: 'erreur de saisie',
    },
  };
  it('Should render all components correctly', async () => {
    const { asFragment } = render(
      <Component
        users={users}
        inputs={inputs}
        onChange={() => {}}
        onClick={() => {}}
        hasSubmitOnce={false}
        isLoading={false}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
    screen.getByRole('textbox', { name: /Compte github/ });
    screen.getByRole('textbox', { name: /Nom d'utilisateur/ });
    expect(screen.queryByRole('alert')).not.toBeInTheDocument();
    screen.getByRole(/list/, { name: 'liste comptes' });
  });

  it('should render loading message when loading', () => {
    render(
      <Component users={[]} inputs={inputs} onChange={() => {}} onClick={() => {}} hasSubmitOnce={false} isLoading />,
    );
    screen.getByRole('textbox', { name: /Compte github/ });
    screen.getByRole('textbox', { name: /Nom d'utilisateur/ });
    screen.getByRole('alert');
    expect(screen.queryByRole(/list/, { name: 'liste comptes' })).not.toBeInTheDocument();
  });

  it('should shows messages when form has submitted once', () => {
    render(
      <Component
        users={users}
        inputs={inputs}
        onChange={() => {}}
        onClick={() => {}}
        hasSubmitOnce
        isLoading={false}
      />,
    );
    expect(screen.getByRole('alert', { name: /githubAccountAlert/ })).toBeInTheDocument();
    expect(screen.getByRole('alert', { name: /errorUsername/ })).toBeInTheDocument();
  });
});
