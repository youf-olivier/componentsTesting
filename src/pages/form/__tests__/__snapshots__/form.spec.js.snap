// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form test suite Should render all components correctly 1`] = `
<DocumentFragment>
  <form
    class="form"
    role="search"
  >
    <div
      class="form__line"
    >
      <label
        class="form__label"
        for="githubAccount"
      >
        Compte github
      </label>
      <div
        class="form__input-container"
      >
        <input
          aria-errormessage="githubAccountAlert"
          id="githubAccount"
          name="githubAccount"
          type="text"
          value="test"
        />
      </div>
    </div>
    <div
      class="form__line form__separator"
    >
      <span>
        Ou
      </span>
    </div>
    <div
      class="form__line"
    >
      <label
        class="form__label"
        for="username"
      >
        Nom d'utilisateur
      </label>
      <div
        class="form__input-container"
      >
        <input
          aria-errormessage="errorUsername"
          id="username"
          name="username"
          type="text"
          value="Jean"
        />
      </div>
    </div>
    <div
      class="form__line form__line--alone"
    >
      <button
        class="btn btn--search"
        type="button"
      >
        Rechercher
      </button>
    </div>
  </form>
  <div
    aria-label="liste comptes"
    class="user-list"
    role="list"
  >
    <div
      aria-label="carte utilisateur"
      class="user-card"
      role="listitem"
    >
      <a
        class="user-card__link"
        href="http://comptegithub.com"
      >
        <img
          alt="login1"
          class="user-card__avatar"
          src="http://avatar1.jpg"
        />
        <span
          class="user-card__login"
        >
          login1
        </span>
      </a>
    </div>
    <div
      aria-label="carte utilisateur"
      class="user-card"
      role="listitem"
    >
      <a
        class="user-card__link"
        href="http://comptegithub.com"
      >
        <img
          alt="login2"
          class="user-card__avatar"
          src="http://avatar2.jpg"
        />
        <span
          class="user-card__login"
        >
          login2
        </span>
      </a>
    </div>
    <div
      aria-label="carte utilisateur"
      class="user-card"
      role="listitem"
    >
      <a
        class="user-card__link"
        href="http://comptegithub.com"
      >
        <img
          alt="login3"
          class="user-card__avatar"
          src="http://avatar3.jpg"
        />
        <span
          class="user-card__login"
        >
          login3
        </span>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
